### DISCUSSION TITLE
`makeRecorderKit` vs `makeERecorderKit`

### DISCUSSION THREAD

**Discussion by tgrecojs** [Agoric/agoric-sdk#8065] [Q&A] [answered]
I've recently begun using recorder kits in an effort to achieve durable pub/sub operations. As the title suggests, I'm seeking clarity regarding the use of `makeRecorderKit` versus `makeERecorderKit`. Specifically, I'm wondering when the use of one is preferred over the other, and vice-versa. 

I'm currently only using `makeRecorderKit` in my code which I'm obtaining via the `prepareRecorderKitMakers` function shown below.

https://github.com/Agoric/agoric-sdk/blob/6f740a40710dd9f1cff8ea8dddc88cf9d6601486/packages/zoe/src/contractSupport/recorder.js#L215-L237

The reason for using `prepareRecorderKitMakers` stems from observing the approach taken in the inter-protocol code shown below. That said, I'm also curious as to how inter's codebase makes use of it (and the reasoning behind doing so).

https://github.com/Agoric/agoric-sdk/blob/6f740a40710dd9f1cff8ea8dddc88cf9d6601486/packages/inter-protocol/src/vaultFactory/vaultFactory.js#L90-L93

Thanks! 🙂

**Comment by dckc** [comment]
@turadg do you have a handy answer for this one?

**Comment by turadg** [comment, marked as answer]
use `makeRecorderKit` unless you have a need specifically for an `EventualRecorderKit`.

https://github.com/Agoric/agoric-sdk/blob/fc69584bb10559dc071ec250f066b941bfe25e1d/packages/zoe/src/contractSupport/recorder.js#L160-L168

https://github.com/Agoric/agoric-sdk/blob/fc69584bb10559dc071ec250f066b941bfe25e1d/packages/zoe/src/contractSupport/recorder.js#L29-L38
